### PR TITLE
Implement models for text response question

### PR DESCRIPTION
**Pull request by BenMQ**

### CODE REVIEW

**Comment by lowjoel** [comment]
@BenMQ can you please clean up all your whitespace? align all the hashes and everything...

File: app/models/course/assessment.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ class Course::Assessment < ActiveRecord::Base
   has_many :multiple_response_questions,
            through: :questions, source: :actable,
            source_type: Course::Assessment::Question::MultipleResponse.name
+  has_many :text_response_questions,
+           through: :questions, source: :actable,
+            source_type: Course::Assessment::Question::TextResponse.name
```
Comment (lowjoel) [inline app/models/course/assessment.rb]: Please fix the whitespace.

File: app/models/course/assessment/answer/text_response.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Course::Assessment::Answer::TextResponse < ActiveRecord::Base
+  acts_as :answer, class_name: Course::Assessment::Answer.name, inverse_of: :actable
+
```
Comment (lowjoel) [inline app/models/course/assessment/answer/text_response.rb]: ...whitespace...

File: app/models/course/assessment/question/text_response.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Course::Assessment::Question::TextResponse < ActiveRecord::Base
+  acts_as :question, class_name: Course::Assessment::Question.name, inverse_of: :actable
+
+  has_many :solutions, class_name: Course::Assessment::Question::TextResponseSolution.name,
+           dependent: :destroy, foreign_key: :question_id, inverse_of: :question
```
Comment (lowjoel) [inline app/models/course/assessment/question/text_response.rb]: does foreign_key need to be specified?

File: app/models/course/assessment/question/text_response_solution.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Course::Assessment::Question::TextResponseSolution < ActiveRecord::Base
+  enum solution_type: [:exact_match, :keyword]
+
+  # validate grade <= max grade?
```
Comment (lowjoel) [inline app/models/course/assessment/question/text_response_solution.rb]: Please clean up stray comments, unless they're meant to be explicit TODOs

File: spec/models/course/assessment/question/text_response_solution_spec.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Course::Assessment::Question::TextResponseSolution, type: :model do
+  it 'belongs to question' do
+    expect(subject).to belong_to(:question)
```
Comment (lowjoel) [inline spec/models/course/assessment/question/text_response_solution_spec.rb]: I prefer trailing periods.

File: app/models/course/assessment.rb (hunk 1 above)
Comment (BenMQ) [inline app/models/course/assessment.rb]: Done.

File: app/models/course/assessment/answer/text_response.rb (hunk 2 above)
Comment (BenMQ) [inline app/models/course/assessment/answer/text_response.rb]: Done.

File: app/models/course/assessment/question/text_response.rb (hunk 3 above)
Comment (BenMQ) [inline app/models/course/assessment/question/text_response.rb]: Actually not sure, I was following what `Course::Assessment::Question::MultipleResponse` does

File: app/models/course/assessment/question/text_response_solution.rb (hunk 4 above)
Comment (BenMQ) [inline app/models/course/assessment/question/text_response_solution.rb]: Added actual validation instead. Was meant as TODO (so `# TODO`: in the future?)

File: spec/models/course/assessment/question/text_response_solution_spec.rb (hunk 5 above)
Comment (BenMQ) [inline spec/models/course/assessment/question/text_response_solution_spec.rb]: Done.

File: app/models/course/assessment/question/text_response_solution.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class Course::Assessment::Question::TextResponseSolution < ActiveRecord::Base
+  enum solution_type: [:exact_match, :keyword]
+
+  validate :validate_grade
+
+  belongs_to :question, class_name: Course::Assessment::Question::TextResponse.name,
+             inverse_of: :solutions
```
Comment (BenMQ) [inline app/models/course/assessment/question/text_response_solution.rb]: not sure if I understand this violation...

**Comment by BenMQ** [comment]
Hmm why does the test fail? It's passing locally.

File: db/schema.rb [hunk 7]
```diff
@@ -178,6 +182,17 @@
     t.integer "grade"
   end
 
+  create_table "course_assessment_question_text_responses", force: :cascade do |t|
+  end
+
+  create_table "course_assessment_question_text_response_solutions", force: :cascade do |t|
+    t.integer "question_id",   null: false, index: {name: "fk__course_assessment_text_response_solution_question"}, foreign_key: {references: "course_assessment_question_text_responses", name: "fk_course_assessment_questi_2fbeabfad04f21c2d05c8b2d9100d1c4", on_update: :no_action, on_delete: :no_action}
```
Comment (lowjoel) [inline db/schema.rb]: where did this name come from? o.O I usually manually truncate the name by removing some common prefix.

File: db/migrate/20151021014315_create_course_assessment_question_text_response.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+class CreateCourseAssessmentQuestionTextResponse < ActiveRecord::Migration
+  def change
+    create_table :course_assessment_question_text_responses do
+    end
+
+    create_table :course_assessment_question_text_response_solutions do |t|
+      t.references :question,
+                   null: false,
+                   index: {
+                     name: :fk__course_assessment_text_response_solution_question
```
Comment (lowjoel) [inline db/migrate/20151021014315_create_course_assessment_question_text_response.rb]: the index should be unnecessary. can you check?

File: db/schema.rb (hunk 7 above)
Comment (BenMQ) [inline db/schema.rb]: I was following this .. https://github.com/Coursemology/coursemology2/blob/master/db/migrate/20150726062900_create_course_assessment_question_multiple_response.rb

File: db/migrate/20151021014315_create_course_assessment_question_text_response.rb (hunk 8 above)
Comment (BenMQ) [inline db/migrate/20151021014315_create_course_assessment_question_text_response.rb]: same as above, was following https://github.com/Coursemology/coursemology2/blob/master/db/migrate/20150726062900_create_course_assessment_question_multiple_response.rb so not too sure :/

File: db/schema.rb (hunk 7 above)
Comment (lowjoel) [inline db/schema.rb]: Probably a length limit. Then it’s fine.

**Pull request merged**